src/Credentials/UserRefreshCredentials.php

```diff
@@ -85,16 +91,19 @@ public function __construct(
             'scope' => $scope,
             'tokenCredentialUri' => self::TOKEN_CREDENTIAL_URI,
         ]);
+        if (array_key_exists('quota_project', $jsonKey)) {
+            $this->quotaProject = (string) $jsonKey['quota_project'];
+        }
         if ($jsonKey['client_id'] === self::CLOUD_SDK_CLIENT_ID
+            && is_null($this->quotaProject)
             && getenv(self::SUPPRESS_CLOUD_SDK_CREDS_WARNING_ENV) !== 'true') {
             trigger_error(
                 'Your application has authenticated using end user credentials '
-                . 'from Google Cloud SDK. We recommend that most server '
-                . 'applications use service accounts instead. If your '
-                . 'application continues to use end user credentials '
-                . 'from Cloud SDK, you might receive a "quota exceeded" '
-                . 'or "API not enabled" error. For more information about '
-                . 'service accounts, see '
+                . 'from Google Cloud SDK. We recommend setting the '
```

Comment:
I think, gcloud sdk will automatically add `quota_project` field and user should not do anything about it.

This means that, probably the warning message is something like "use more recent version of gcloud sdk" or something.

@broady 

Do you have any advise on the warning message here?

Reply:
Ahh yes, that makes sense. "Please upgrade to the latest version of Google Cloud SDK"

Reply:
Yeah, and depending on the gcloud sdk's implementation, you'll also have to re-create the auth file (I'm not sure how it's implemented).

Reply:
Great question, I think we should point people to a page on c.g.c that gives you resolution steps.

It's probably "upgrade gcloud and run `gcloud auth application-default login` again"

We will also have a command like `gcloud auth application-default set-quota-project XXX`.

Reply:
Hold off on changing the error message to something that references gcloud. There'll be a separate timeline for updating docs, samples, tutorials, error messages.

Reply:
(I'd just revert this block)